wkhtmltopdf-image/src/main/docker/app.py

```diff
@@ -0,0 +1,123 @@
+#! /usr/bin/env python
+import json
+import os
+import tempfile
+
+from executor import execute
+from werkzeug.exceptions import HTTPException, NotFound
+from werkzeug.routing import Map, Rule
+from werkzeug.wrappers import Request, Response
+from werkzeug.wsgi import wrap_file
+
+"""
+This python server is responsible for generating PDFs from HTML
+using an underlying wkhtmltopdf command-line tool.
+
+To use this application, the user must send a POST request with
+base64 or form encoded encoded HTML content and the wkhtmltopdf Options in
+request data, with keys 'contents' and 'options'.
+The application will return a response with the PDF file.
+"""
+class WkHtmlToPdfServer():
+    def __init__(self):
+        self.url_map = Map([
+            Rule('/', endpoint='generate_pdf'),
+            Rule('/health', endpoint='health_check')
+        ])
+
+    """
+    Generate a PDF from submitted HTML.
+    """
+    def on_generate_pdf(self, request):
+        if request.method != 'POST':
+            return
+
+        request_is_json = request.content_type.endswith('json')
+
+        with tempfile.NamedTemporaryFile(suffix='.html') as source_file:
+
+            if request_is_json:
+                # If a JSON payload is there, all data is in the payload
+                payload = json.loads(request.data)
+                source_file.write(payload['contents'].decode('base64'))
+                options = payload.get('options', {})
+            elif request.files:
+                # First check if any files were uploaded
+                source_file.write(request.files['file'].read())
+                # Load any options that may have been provided in options
+                options = json.loads(request.form.get('options', '{}'))
+
+            source_file.flush()
+
+            # Evaluate argument to run with subprocess
+            args = ['wkhtmltopdf']
+
+            # Add Global Options
+            if options:
+                for option, value in options.items():
+                    args.append('--%s' % option)
+                    if value:
+                        args.append('"%s"' % value)
+
+            # Add source file name and output file name
+            file_name = source_file.name
+            args += [file_name, file_name + ".pdf"]
+
+            # Execute the command using executor
+            execute(' '.join(args))
```

Comment:
For streaming:
cat FILE | wkhtmltopdf [OPTIONS] - -
gets you the pdf to stdout.
the first dash is to use sdtin and second to use stdout